Islandora/documentation
# Fixing docs.yml to point to missing pages

**GitHub Issue**: https://github.com/Islandora/documentation/issues/1352

# What does this Pull Request do?

Updates mkdcos.yml to include a number of pages that were missing from the index but exist in our docs folder. Rearranged something under grouped headings ot try to make the docs more navigable. 

As a bonus, added a missing screenshot from the Collections documentation.

# What's new?
- A screenshot on the Collections docs (asset was missing)
- Links to:
      - alpaca/alpaca-technical-stack.md
      - contributing/hacking-on-islandora.md
      - islandora/drupal-bundle-configurations.md
      - technical-documentation/checking-coding-standards.md
      - technical-documentation/drupal-project.md
      - technical-documentation/resizing_vm.md
      - user-documentation/intro-to-ld-for-islandora-8.md
      - user-documentation/versioning.md 
- New subheadings to group similar pages in the docs:
   - Content in Islandora 8
   - How-To
   - Integrations
   - Tests
   - Alpaca
   - Virtual Machines

# How should this be tested?

Look at the new docs build and verify that links go to the things they should go to and I haven't broken anything.

# Additional Notes:
There are a number of pages from the list in #1352 that I haven't added here because they don't really have a place in the official docs. I'll address those in a separate pull request.

# Interested parties
@Islandora/8-x-committers
